spec/src/main/asciidoc/core/beanmanager_lite.asciidoc

```diff
@@ -24,10 +23,11 @@ Thus, any bean may obtain an instance of `BeanContainer` by injecting it:
 
 [source, java]
 ----
-@Inject BeanContainer manager;
+@Inject BeanContainer container;
 ----
 
 The operations of `BeanContainer` may be called at any time during the execution of the application.
+// TODO Full has restrictions on when BeanManager methods can be called, do we want to reflect them here in some way?
```

Comment:
How about we instead change the `The BeanManager object` and mention that those rules apply to both, BM and `BeanContainer`?

Reply:
That should be implied by

> In {cdi_full} environment, `BeanContainer` is subject to the same rules as `BeanManager`.

What I'm more thinking about here is: if we're saying that in Lite, `BeanContainer` can be invoked at any time, that's a superset of what's allowed in Full. Which might be problematic at least from the "Lite is a strict subset of Full" perspective.

Reply:
Or at least, it might _look_ like a superset. That will depend on how we define the container lifecycle in Lite, which isn't there yet. (In Full, it's defined in the SPI section.)

Reply:
> BeanContainer can be invoked at any time

That's not what you wrote there though. You stated `... at any time during the execution of the application` which means after the CDI container is bootstrapped, doesn't it? In other words, in Lite you cannot even try to use `BeanContainer` earlier.
And so long as you cannot use `BeanContainer` where you couldn't use BM (from within extensions), you should be good.

Or am I missing something?

Reply:
Again, "at any time during the execution of the application" depends on how we define lifecycle, which we didn't do yet. If we define it in a way that all initialization phases are considered _before_ application execution, then sure.

I added the TODO here mainly as a reminder, to revisit this section after we define lifecycle. I could have worded it more explicitly, for sure -- sorry about that.